src/Synth/ADnote.cpp

```diff
@@ -1805,19 +1808,29 @@ inline void ADnote::ComputeVoiceOscillatorWaveTableModulation(int nvoice)
                  NoteVoicePar[nvoice].basefuncpar));
 
             float semantic = par * 512.f;
-            if(semantic >= wt->size_semantics())
-                semantic = wt->size_semantics() - 1;
+            if(semantic >= wt->size_semantics() - 1)
```

Comment:
Is `-1` not a valid array index?

Reply:
for wt->getWaveAt(freq, semantic+1) i needed to limit semantic further, but this is surely not the right place.

Reply:
changed it back and check at the point where i use the value